src/lib.rs

```diff
@@ -0,0 +1,187 @@
+//! Crate extends the list of possible sources provided by [config-rs](https://github.com/mehcode/config-rs).
+//! Crate provides an asynchronous `RedisSource` source using the [redis-rs](https://github.com/redis-rs/redis-rs).
+//! `RedisSource` supports reading configuration:
+//! - from Hash using the HGETALL command,
+//! - from String using the GET command,
+//!
+//! # Optional Features
+//!
+//! There are a few features defined in [redis-rs](https://github.com/redis-rs/redis-rs) that can enable additional functionality if so desired.
+//! Some of them are turned on by default.
+//! - ahash: enables ahash map/set support & uses ahash internally (+7-10% performance) (optional)
+//! - tokio-comp: enables support for tokio runtime (enabled by default)
+//! - async-std-comp: enables support for async-std runtime (optional)
+//!
+//! Tls features
+//! - async-std-native-tls-comp: enables support for native-tls for async-std (optional)
+//! - async-std-rustls-comp: enables support for rustls for async-std (optional)
+//! - tokio-native-tls-comp: enables support for native-tls for tokio (optional)
+//! - tokio-rustls-comp: enables support for rustls for tokio (optional)
+//!
+//! See the examples for general usage information.
+
+mod errors;
+
+pub use errors::{SourceError, SourceResult};
+
+use std::{collections::HashMap, fmt::Debug};
+
+use config::{AsyncSource, ConfigError, Map};
+use redis::AsyncCommands;
+
+const REDIS_CONFIG_URL_KEY: &str = "REDIS_CONFIG_URL";
+
+/// A configuration async source backed up by a Redis.
+///
+/// It supports retrieving configuration in JSON format from Strings and Hashes types in Redis.
+///
+/// # Examples
+///
+/// ```ignore
```

Comment:
is it possible to un-ignore this doctest?
(I failed to quickly find a solutions for compiling ```let config = get_config().await.unwrap();``` here (the last line), but maybe you will have more time and willingness)

Reply:
Fixed, example was incorrect. Changed to no_run for this doc test, to not run integration tests in doc tests.

But you can run it locally with docker

`docker run --name bootcamp_redis -p 6379:6379 -d redis`